docs: add page query guide feedback improvements (#17929)
gatsbyjs/gatsby: docs/docs/page-query.md

}
```

> Below our `HomePage` component declaration, export a new constant called `query`. The name of the constant isn't important, Gatsby just looks for an exported `graphql` string from the file. Note that you can only have one page query per file.

Reviewer: some how i am confused with the wording `page query` there - it is never used before in this doc...

PS: only the file name of the MD is `page-query.md`

> Below our `HomePage` component declaration, export a new constant called `query`. The name of the constant isn't important, Gatsby just looks for an exported `graphql` string from the file. Note that you can only have one page query per file.

Reviewer: i see it's later also used, maybe we introduce a small definition at the beginning. for example distinction between a `normal` query and then that is there a `page query` and thats why this document explains it ...

> Below our `HomePage` component declaration, export a new constant called `query`. The name of the constant isn't important, Gatsby just looks for an exported `graphql` string from the file. Note that you can only have one page query per file.

Reviewer 2: Taking the opportunity to differentiate between types of queries is always a good idea! This one should definitely focus on page query since that's what the doc is about, so any definitions should be brief.

## How to add query variables to a page query

Variables can be added to page queries (not static queries) through the context object that is an argument of the [`createPage` API](/docs/actions/#createPage).

Consider the following query:

```js:title=src/templates/blog-post.js
export const query = graphql`
  query MdxBlogPost {
    mdx(title: { eq: "Using a Theme" }) {
      id
      title
    }
  }
`
```

The `MdxBlogPost` query will return an MDX node that has been created in a site where `gatsby-plugin-mdx` is installed and `.mdx` files have been [sourced](/docs/content-and-data/) with `gatsby-source-filesystem`, so long as it matches the argument passed in that says the `title` must equal (`eq`) the string `"Using a Theme"`.

In addition to hardcoding an argument directly into the query, you can pass in a variable. The query can be changed to include a variable like this:

```js:title=src/templates/blog-post.js
export const query = graphql`
  query MdxBlogPost($title: String) { // highlight-line
    mdx(title: {eq: $title}) { // highlight-line
      id
      title
    }
  }
`
```

> Then when a page is created from this blog post template, you can provide an object as part of the page's context. Keys in the context object that match up with arguments in the page query (in this case: `"title"`), will be used as variables. Variables are prefaced with `$`, so passing a `title` property will become `$title` in the query.

Reviewer 2: ```suggestion
When a page is created dynamically from this blog post template in `gatsby-node.js`, you can provide an object as part of the page's context. Keys in the context object that match up with arguments in the page query (in this case: `"title"`), will be used as variables. Variables are prefaced with `$`, so passing a `title` property will become `$title` in the query.
```

```js:title=gatsby-node.js
posts.forEach(({ node }, index) => {
  createPage({
    path: node.fields.slug,
    component: path.resolve(`./src/templates/blog-post.js`),
    // values in the context object are passed in as variables to page queries
    context: {
      title: node.title, // "Using a Theme"
    },
  })
})
```

> The docs have more information about [creating pages programmatically](/docs/programmatically-create-pages-from-data/).

Reviewer 2: ```suggestion
For more information, check out the docs on [creating pages programmatically](/docs/programmatically-create-pages-from-data/).
```
